### PR TITLE
Latest pairs added on Uniswap by Eloise

### PR DESCRIPTION
**Pull request by Eloise1988**
Thanks for submitting this tutorial for Figment Learn!

Before you hit "Create Pull Request" make sure you have read the [Tutorial Contribution Guidelines](https://www.notion.so/figmentnetworks/Figment-Learn-Contributor-Guidelines-b08691366e144d33a5636d9276a44ec8).
If you do not follow these guidelines, it might take longer for us to review your tutorial and get it published on Learn.

- [x] My tutorial follows the [Tutorial Contribution Guidelines](https://www.notion.so/figmentnetworks/Figment-Learn-Contributor-Guidelines-b08691366e144d33a5636d9276a44ec8).

In particular:

- [x] I've followed the [tutorial template](https://www.notion.so/figmentnetworks/Sample-Tutorial-Structure-667ac2aad36f4f94a3ebaca053180b2d) structure.
- [x] I've proofread the tutorial for spelling and grammar mistakes.
- [x] I've run a linter & prettifier to format my code blocks.
- [x] I've thoroughly explained the code, with inline code comments and text explanations.
- [x] I've run the tutorial to test it and the code works.

### CODE REVIEW

File: network-documentation/thegraph/tutorials/uniswap_latestpairs.md [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+
+To get the data in Postman, you will need to:
+1. Insert in the URL box the HTTP Query url from the Uniswap V3 subgraph (https://api.thegraph.com/subgraphs/name/uniswap/uniswap-v3)
+2. Change the GET into a POST request
+3. Copy/paste the above GraphQL query in Body and select GraphQL
+4. Click Send
+
+You should get the same JSON table as on TheGraph.
+
+<img width="1100" alt="potman_query" src="https://user-images.githubusercontent.com/53000607/133658202-670210a1-4509-4754-a164-1e22514d4594.png">
```
Comment (zurgl) [inline network-documentation/thegraph/tutorials/uniswap_latestpairs.md:110]: All images have to be part of the PR

File: network-documentation/thegraph/tutorials/uniswap_latestpairs.md [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+
+
+### Connecting the model through Google Sheet
+Once you confirmed that the query is functioning in Postman, you can transform the code into a Javascript Fetch request (for Google Sheet) using the Code button underneath the Save button.
+
+<img width="977" alt="javascript_postman" src="https://user-images.githubusercontent.com/53000607/133802683-9462b554-06f8-4d52-9533-35491102f006.png">
```
Comment (zurgl) [inline network-documentation/thegraph/tutorials/uniswap_latestpairs.md:116]: All images have to be part of the PR

**Review by zurgl — CHANGES_REQUESTED** [review]
Hi @Eloise1988 

Thanks for writing this tutorial! It's pretty original, this is the first time I see something like that and I really appreciate it. Please review my comments.

File: network-documentation/thegraph/tutorials/uniswap_latestpairs.md [hunk 3]
```diff
@@ -0,0 +1,261 @@
+# Scan the latest pairs added on UniswapV3 using TheGraph in Google Sheets 
+[<img width="1381" alt="gif_uni" src="https://user-images.githubusercontent.com/53000607/133595220-6e918166-cfc9-4d9c-9b5b-5bc4395852ce.gif">](https://docs.google.com/spreadsheets/d/1tME9nMh79KzZP4Wmld7lezom6je4BOw_0T9ABf5GKXE/edit?usp=sharing)
```
[the remark this answers is not in the document]
Comment (Eloise1988) [reply network-documentation/thegraph/tutorials/uniswap_latestpairs.md]: I changed it to the following form :
UNISWAP

File: network-documentation/thegraph/tutorials/uniswap_latestpairs.md [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+
+# Introduction
+This tutorial is built to help non-technical people get a sense of how to interact with TheGraph and connect on-chain data into Google Sheets. I've been working on gathering crypto data in Google Sheets for a couple of years now and I found that sheets is a useful interface when filtering for new listed pairs on decentralized exchanges. In this tutorial you'll **learn how to find a UniswapV3 subgraph on The Graph, make a GraphQL request, translate the query in Javascript (Google Sheet's programming language) using Postman, and finally retrieve the pairs in the Sheet with a user-defined formula.**
+
+### Prerequisites
```
[the remark this answers is not in the document]
Comment (Eloise1988) [reply network-documentation/thegraph/tutorials/uniswap_latestpairs.md]: Changed it to ##

File: network-documentation/thegraph/tutorials/uniswap_latestpairs.md [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+For beginners with basic coding skills. This training assumes that you have a Gmail/Google account as we will be building upon Google Sheets. Also it would help if you have access to Postman to get an easier view on how to test a GraphQL request from TheGraph and transform it into Javascript code which is  the Official programming language of Google Sheet . 
+
+* **GraphQL & Javascript knowledge are a plus.** Code can be written directly with the help of Postman and The Graph's website.
+* **Postman's** software makes API development easier.
+* Code :
+  * uniswap.gs - for retrieving Uniswap's latest pair through The Graph API (later explained in the tutorial)
+  * importJson.gs - for working with JSON objects in Google Sheets (later explained in the tutorial)
+
+
+### First, what is Uniswap?
```
[the remark this answers is not in the document]
Comment (Eloise1988) [reply network-documentation/thegraph/tutorials/uniswap_latestpairs.md]: Changed it to ##

File: network-documentation/thegraph/tutorials/uniswap_latestpairs.md [hunk 6]
```diff
@@ -0,0 +1,261 @@
...
+[Uniswap](https://academy.binance.com/en/articles/what-is-uniswap-and-how-does-it-work) is a **decentralized exchange protocol built on Ethereum.** To be more precise, it is an automated liquidity protocol. There is no order book or any centralized party required to make trades. Uniswap allows users to trade without intermediaries, with a high degree of decentralization and censorship-resistance.
+
+#### This is how Uniswap’s Decentralized Exchange looks like:
+
+[<img src=https://user-images.githubusercontent.com/53000607/132863640-4889463d-0e54-4a9e-b7ca-3a71017f8fc7.png width="450">](https://uniswap.exchange/swap)   <img src=https://user-images.githubusercontent.com/53000607/132863582-dd3c9ea6-e1e4-43f2-b42b-c27254631006.png width="550">
```
[the remark this answers is not in the document]
Comment (Eloise1988) [reply network-documentation/thegraph/tutorials/uniswap_latestpairs.md]: I changed it to the following form :
UNISWAPEXCHANGE

File: network-documentation/thegraph/tutorials/uniswap_latestpairs.md [hunk 7]
```diff
@@ -0,0 +1,261 @@
...
+
+### Getting familiar with TheGraph and GraphQL queries
+[The Graph](https://thegraph.com/) is a **decentralized protocol for indexing and querying data** from blockchains. It is able to query networks like Ethereum and since Uniswap is built on Ethereum, it will allow us to get its on-chain data. 
+ 
+#### Finding Uniswap V3 subgraph on Thegraph
+In this tutorial, we will be focusing on getting blockchain data on Version 3 of Uniswap. All you need to do is to search in TheGraph's explorer bar for Uniswap V3. The following picture shows you what TheGraph looks like and which subgraph we will be using:
+
+* https://thegraph.com/legacy-explorer/subgraph/uniswap/uniswap-v3
+
+[<img width="490" alt="thegraph" src="https://user-images.githubusercontent.com/53000607/133580577-56cecc0a-79c3-473d-83f9-6e9420ec6afd.png">](https://thegraph.com/legacy-explorer/) [<img width="510" alt="thegraph_uni2" src="https://user-images.githubusercontent.com/53000607/132865398-6227fe0c-d447-408d-9e67-5767d8125744.png">](https://thegraph.com/legacy-explorer/subgraph/uniswap/uniswap-v3)
```
[the remark this answers is not in the document]
Comment (Eloise1988) [reply network-documentation/thegraph/tutorials/uniswap_latestpairs.md]: I changed the images in a gif
UNISWAPv3SUBGRAPH

File: network-documentation/thegraph/tutorials/uniswap_latestpairs.md [hunk 8]
```diff
@@ -0,0 +1,261 @@
...
+
+#### Building the GraphQL query
+[GraphQL](https://en.wikipedia.org/wiki/GraphQL) is an **open-source data query and manipulation language for APIs,** and a runtime for fulfilling queries with existing data. GraphQL was developed internally by Facebook in 2012 before being publicly released in 2015.
```
[the remark this answers is not in the document]
Comment (Eloise1988) [reply network-documentation/thegraph/tutorials/uniswap_latestpairs.md:36]: I changed to https://graphql.org/

File: network-documentation/thegraph/tutorials/uniswap_latestpairs.md [hunk 9]
```diff
@@ -0,0 +1,261 @@
...
+
+#### Building the GraphQL query
+[GraphQL](https://en.wikipedia.org/wiki/GraphQL) is an **open-source data query and manipulation language for APIs,** and a runtime for fulfilling queries with existing data. GraphQL was developed internally by Facebook in 2012 before being publicly released in 2015. 
+
+On TheGraph interface in playgroud mode, there is an example of a GraphQL query. This query gets the first 5 factories with id, poolCount, txCount and totalVolumeUSD, as well as the first 5 bundles with their id and ethPriceUSD.
+
+<img width="660" alt="std_graph_query" src="https://user-images.githubusercontent.com/53000607/133599485-8dd9fa85-f20c-4c29-9f04-cd843d593511.png"> <img width="340" alt="parameters_graph" src="https://user-images.githubusercontent.com/53000607/133599807-0d6c666f-5a02-44ca-a70c-fbf5e485fb64.png">
```
[the remark this answers is not in the document]
Comment (Eloise1988) [reply network-documentation/thegraph/tutorials/uniswap_latestpairs.md]: GIF
PLAYGROUND

File: network-documentation/thegraph/tutorials/uniswap_latestpairs.md [hunk 10]
```diff
@@ -0,0 +1,261 @@
...
+
+[<img width="630" alt="thegraph_code" src="https://user-images.githubusercontent.com/53000607/132865391-1d131a43-7973-47d1-a182-a4fb5bfec97c.png">](https://thegraph.com/legacy-explorer/subgraph/uniswap/uniswap-v3)  <img width="370" alt="parameters_graph" src="https://user-images.githubusercontent.com/53000607/133599807-0d6c666f-5a02-44ca-a70c-fbf5e485fb64.png">
+
+### Testing model & translating the query into javascript using Postman
+
+* [Download Postman](https://www.postman.com/downloads/)
+
+Once you download Postman, you can open a new window like shown below:
+
+<img width="1437" alt="postmanclean" src="https://user-images.githubusercontent.com/53000607/133645104-1099b76f-41cd-4637-94e9-79b965dcce92.png">
```
[the remark this answers is not in the document]
Comment (Eloise1988) [reply network-documentation/thegraph/tutorials/uniswap_latestpairs.md]: IMAGE:
PLAYGROUNDCONSTRAINTS
POSTMANCLEAN

File: network-documentation/thegraph/tutorials/uniswap_latestpairs.md (hunk 1 above)
Comment (Eloise1988) [reply network-documentation/thegraph/tutorials/uniswap_latestpairs.md]: IMAGE:
POSTMANQUERY

File: network-documentation/thegraph/tutorials/uniswap_latestpairs.md [hunk 11]
```diff
@@ -0,0 +1,261 @@
...
+      };
+
+      // [4] Calling the URLfetch using the ImportJSONAdvanced function (by Brad Jasper and Trevor Lohrbeer version 1.5) that was created for Google sheets
+      return ImportJSONAdvanced('https://api.thegraph.com/subgraphs/name/uniswap/uniswap-v3',requestOptions,'','noInherit',includeXPath_,defaultTransform_);
+}
+```
+#### Copy/Pasting the code in Google Sheet
+Open a Google sheet where you wish to use the Uniswap function -> Go to **Tools › Script editor**
+
+<img width="330" alt="gs1" src="https://user-images.githubusercontent.com/53000607/133906614-c8cf356d-6fa2-4440-8e10-1b42b9e5f540.png"><img width="350" alt="gs2" src="https://user-images.githubusercontent.com/53000607/133906613-2aba9315-8328-47c3-b0d6-0e17f98f50cc.png"><img width="330" alt="gs3" src="https://user-images.githubusercontent.com/53000607/133906612-438908c5-2e53-4dbb-aadf-86ebd61e8ffd.png">
```
[the remark this answers is not in the document]
Comment (Eloise1988) [reply network-documentation/thegraph/tutorials/uniswap_latestpairs.md]: GIF CREATED:
GOOGLESHEETEDITOR

File: network-documentation/thegraph/tutorials/uniswap_latestpairs.md [hunk 12]
```diff
@@ -0,0 +1,261 @@
...
+
+```javascript
+/**
+* @OnlyCurrentDoc
+*/
+```
+
+When running the code, you will probably be asked to give permission to the code. Here are the steps:
+
+[<img width="32%" alt="goog_auth_5" src="https://user-images.githubusercontent.com/53000607/132861811-0d7c4712-8f8c-4f4b-892c-2779a4035036.png"> <img width="32%" alt="goog_auth_4" src="https://user-images.githubusercontent.com/53000607/132861818-d9d927d6-c230-4924-9c35-1bf528afbe72.png"> <img width="32%" alt="goog_auth_3" src="https://user-images.githubusercontent.com/53000607/132861821-62440a1f-99b3-4891-80a0-3f6b2c6365d3.png"> <img width="32%" alt="goog_auth_2" src="https://user-images.githubusercontent.com/53000607/132861825-6da9adbc-6bf2-4733-bf9b-4d5476b8f19f.png"> <img width="32%" alt="goog_auth" src="https://user-images.githubusercontent.com/53000607/132861831-8dbba6ee-617f-44ec-938c-7a922b498f76.png"> <img width="32%" alt="postman" src="https://user-images.githubusercontent.com/53000607/132861836-9fe4bd08-9ad1-42ee-893d-70c89d9d9dd8.png">](https://developers.google.com/apps-script/guides/services/authorization)
```
[the remark this answers is not in the document]
Comment (Eloise1988) [reply network-documentation/thegraph/tutorials/uniswap_latestpairs.md]: I took out that part, it will be less confusing

**Comment by zurgl** [comment]
@Eloise1988 Thanks for the quick feedback.
All the images should be part of the PR, currently I see only one file in the PR.

For example, for the first image you need to: 
* Copy `https://github.com/Eloise1988/THEGRAPH/blob/main/IMAGES/UNISWAP.gif` to `.gitbook/assets/UNISWAP.gif`
* Change ```UNISWAP``` to ```UNISWAP``` in your markdown file

You need to do it for all the images.

Another requirement to publish you content is that we support only 2 levels for heading
 For example an accepted doc is like.
 ```text
 title 
 -------------------------------------------
 
 # Section 1
 
 lskjdlsjdlskj
 
 ## subSection 1
 
 sldksmldksmldks
 
 # Section 2
 
 sdklsmdk
 
 # Section 3
 
 ## subSection 3
 
 sdklsmdk
 
```
We cannot manage more than two level for headings.
I guess when theses two requirement I meet, i will be able to merge your work.

Thanks for your comprehensions.

**Comment by Eloise1988** [comment]
> @Eloise1988 Thanks for the quick feedback.
> All the images should be part of the PR, currently I see only one file in the PR.
> 
> For example, for the first image you need to:
> 
> * Copy `https://github.com/Eloise1988/THEGRAPH/blob/main/IMAGES/UNISWAP.gif` to `.gitbook/assets/UNISWAP.gif`
> * Change `UNISWAP` to `UNISWAP` in your markdown file
> 
> You need to do it for all the images.
> 
> Another requirement to publish you content is that we support only 2 levels for heading
> For example an accepted doc is like.
> 
> ```
> title 
> -------------------------------------------
> 
> # Section 1
> 
> lskjdlsjdlskj
> 
> ## subSection 1
> 
> sldksmldksmldks
> 
> # Section 2
> 
> sdklsmdk
> 
> # Section 3
> 
> ## subSection 3
> 
> sdklsmdk
> ```
> 
> We cannot manage more than two level for headings.
> When theses two requirements are fulfilled, we will be able to merge your work.
> 
> Do not forget to mark the conversations as resolved when you've made the changes
> 
> Thanks for your comprehensions.

@zurgl Thank you for the feedback. I think this time I understood. I've downloaded all images in the subfolder .gitbook/assets/ and reference them in the tutorial with the corresponding path.

I also changed the headers so that there are on # an ##.
Regards,

**Pull request merged**